API/Routes/Routes.go

```diff
@@ -11,5 +11,6 @@ func RoutesRegistrations(e *echo.Echo) {
 	e.POST("/login", Login)
 	e.GET("/token", Token, middleware.JWT([]byte(os.Getenv("JWT_SECRET"))))
 	e.POST("/offer/create", CreateOffer, middleware.JWT([]byte(os.Getenv("JWT_SECRET"))))
+	e.POST("/offer/save", UpdateOffer, middleware.JWT([]byte(os.Getenv("JWT_SECRET"))))
```

Comment:
Tu devrais plutôt faire :
```suggestion
	e.POST("/offer", CreateOffer, middleware.JWT([]byte(os.Getenv("JWT_SECRET"))))
	e.PATCH("/offer/:id", UpdateOffer, middleware.JWT([]byte(os.Getenv("JWT_SECRET"))))
	e.GET("/offer/:id, GetOffer, middleware.JWT([]byte(os.Getenv("JWT_SECRET")))) // renvoie les infos de l'offre
```

Du coup en faisant ça tu peux supprimer ta structure OfferUpdate et utiliser l'autre pour les deux routes.